examples/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect.ino

```diff
@@ -0,0 +1,591 @@
+void connectToWiFi();
+void initializeAzureIoTHubClient();
+void initializeMQTTClient();
+void connectMQTTClientToAzureIoTHub();
+
+// Telemetry and message-callback functions.
+void onMessageReceived(int messageSize);
+static void sendTelemetry();
+static char* generateTelemetry();
+
+// SAS Token related functions.
+static void generateMQTTPassword();
+static void generateSASBase64EncodedSignedSignature(
+    uint8_t const* sasSignature, size_t const sasSignatureSize,
+    uint8_t* encodedSignedSignature, size_t encodedSignedSignatureSize,
+    size_t* encodedSignedSignatureLength);
+static uint64_t getSASTokenExpirationTime(uint32_t minutes);
+
+// Time and Error functions.
+static unsigned long getTime();
+static String getFormattedDateTime(unsigned long epochTimeInSeconds);
+static String mqttErrorCodeName(int errorCode);
+
+/*---------------------------*/
+/*    Main code execution    */
+/*---------------------------*/
+
+/*
+ * setup:
+ * Initialization and connection of serial communication, WiFi client, Azure IoT SDK for C client, 
+ * and MQTT client.
+ */
+void setup() 
+{
+  while (!Serial);
+  Serial.begin(MBED_CONF_PLATFORM_DEFAULT_SERIAL_BAUD_RATE);
+  pinMode(LED_PIN, OUTPUT);
+
+  digitalWrite(LED_PIN, HIGH);
+
+  connectToWiFi();
+  initializeAzureIoTHubClient();
+  initializeMQTTClient();
+  connectMQTTClientToAzureIoTHub();
+
+  digitalWrite(LED_PIN, LOW);
+}
+
+/*
+ * loop:
+ * Check for connection and reconnect if necessary.
+ * Send Telemetry and receive messages.
+ */
+void loop() 
+{
+  if (WiFi.status() != WL_CONNECTED) 
+  {
+    connectToWiFi();
+  }
+
+  // Telemetry
+  if (millis() > telemetryNextSendTimeMs) 
+  {
+    // Check for MQTT Client connection to Azure IoT hub. Reconnect if needed.
+    if (!mqttClient.connected()) 
+    {
+      connectMQTTClientToAzureIoTHub();
+    }
+
+    sendTelemetry();
+    telemetryNextSendTimeMs = millis() + IOT_CONFIG_TELEMETRY_FREQUENCY_MS;
+  }
+
+  // MQTT loop must be called to process Telemetry and Cloud-to-Device (C2D) messages.
+  mqttClient.poll();
+  delay(500);
+}
+
+/*-----------------------------------------------*/
+/*    Initialization and connection functions    */
+/*-----------------------------------------------*/
+
+/*
+ * connectToWifi:
+ * The WiFi client connects, using the provided SSID and password.
+ * The WiFi client synchronizes the time on the device. 
+ */
+void connectToWiFi() 
+{
+  logString = "Attempting to connect to WIFI SSID: ";
+  LogInfo(logString + IOT_CONFIG_WIFI_SSID);
+
+  while (WiFi.begin(IOT_CONFIG_WIFI_SSID, IOT_CONFIG_WIFI_PASSWORD) != WL_CONNECTED) 
+  {
+    Serial.print(".");
+    delay(IOT_CONFIG_WIFI_CONNECT_RETRY_MS);
+  }
+  Serial.println();
+
+  logString = "WiFi connected, IP address: ";
+  LogInfo(logString + WiFi.localIP() + ", Strength (dBm): " + WiFi.RSSI());
+  LogInfo("Syncing time.");
+
+  while (getTime() == 0) 
+  {
+    Serial.print(".");
+    delay(500);
+  }
+  Serial.println();
+
+  LogInfo("Time synced!");

```

Comment:
Please use logs in this sample to similar to what is used in other existing samples for uniformity:
https://github.com/Azure/azure-sdk-for-c-arduino/blob/main/examples/Azure_IoT_Hub_ESP32/Azure_IoT_Hub_ESP32.ino#L124

If you believe the print should be different, please propose a change to all the samples.

Reply:
Added logging using SerialLogger